src/odemis/odemisd/modelgen.py

```diff
@@ -672,31 +671,43 @@ def _update_properties(self, comp_name):
         comp = self._get_component_by_name(comp_name)
 
         # Initialize persistent properties from settings file
-        props, _ = self.get_persistent(comp)
+        props, _ = self.get_persistent(comp_name)
+        persistent_props_values = self.persistent_props.get(comp_name, {})
         for prop_name in props:
             # If persistent property wasn't saved in settings file, still check for errors
             try:
                 va = getattr(comp, prop_name)
             except AttributeError:
-                raise SemanticError("Error in settings file: "
+                logging.warning("Error in settings file: "
                         "Component '%s' has no property '%s'." % (comp_name, prop_name))
+                del persistent_props_values[prop_name]
+                break
             if not isinstance(va, model.VigilantAttributeBase):
-                raise SemanticError("Error in settings file: "
+                logging.warning("Error in settings file: "
                         "Component '%s' has no property (VA) '%s'." % (comp_name, prop_name))
+                del persistent_props_values[prop_name]
+                break
 
             # Load value from settings file if available
-            if comp_name in self._persistent_props and prop_name in self._persistent_props[comp_name]:
-                value = self._persistent_props[comp_name][prop_name]
+            if prop_name in persistent_props_values:
+                value = persistent_props_values[prop_name]
                 try:
                     va.value = value
                 except Exception as exp:
-                    raise ValueError("Error in settings file: "
+                    logging.warning("Error in settings file: "
                                      "%s.%s = '%s' failed due to '%s'" %
                                      (comp_name, prop_name, value, exp))
+                del persistent_props_values[prop_name]
+
+            # Update persistent_props with va value
+            if not comp_name in self.persistent_props:
+                self.persistent_props[comp_name] = {prop_name: va.value}
```

Comment:
I think you could replace it with a single line, but I'm not sure it's readable:
`self.persistent_props.setdefault(comp_name, {})[prop_name] = va.value`